Microsoft.Toolkit.Uwp.UI.Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,395 @@
+                _tabScrollBackButton.Click += ScrollTabBackButton_Click;
+            }
+
+            if (_tabScrollForwardButton != null)
+            {
+                _tabScrollForwardButton.Click -= ScrollTabForwardButton_Click;
+                _tabScrollForwardButton.Click += ScrollTabForwardButton_Click;
+            }
+        }
+
+        private void ScrollTabBackButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_tabScroller != null)
+            {
+                _tabScroller.ChangeView(Math.Max(0, _tabScroller.HorizontalOffset - SCROLL_AMOUNT), null, null);
+            }
+        }
+
+        private void ScrollTabForwardButton_Click(object sender, RoutedEventArgs e)
+        {
+            if (_tabScroller != null)
+            {
+                _tabScroller.ChangeView(Math.Min(_tabScroller.ScrollableWidth, _tabScroller.HorizontalOffset + SCROLL_AMOUNT), null, null);
+            }
+        }
+
+        private void TabView_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            // We need to do this calculation here in Size Changed as the
+            // Columns don't have their Actual Size calculated in Measure or Arrange.
+            if (_tabViewContainer != null)
+            {
+                // Look for our special columns to calculate size of other 'stuff'
+                var taken = _tabViewContainer.ColumnDefinitions.Sum(cd => GetIgnoreColumn(cd) ? 0 : cd.ActualWidth);
+
+                // Get the column we want to work on for available space
+                var tabc = _tabViewContainer.ColumnDefinitions.FirstOrDefault(cd => GetConstrainColumn(cd));
+                var tabs = GetTabSource();
+
+                if (tabc != null && tabs != null)
+                {
+                    var available = ActualWidth - taken;
+                    var required = 0.0;
+
+                    if (available > 0)
+                    {
+                        // Calculate the width for each tab from the provider and determine how much space they take.
+                        var tvis = _tabItemsPresenter?.FindDescendants<TabViewItem>();
+                        if (tvis != null)
+                        {
+                            var widthIterator = TabWidthProvider.ProvideWidth(tvis, tabs, available, this).GetEnumerator();
+
+                            foreach (var tab in tvis)
+                            {
+                                if (widthIterator.MoveNext())
+                                {
+                                    var width = widthIterator.Current;
+                                    if (width > 0)
+                                    {
+                                        tab.Width = Math.Max(width, 0);
+                                    }
+                                    else
+                                    {
+                                        tab.Width = double.NaN;
```

Comment:
If I specify a width for tabs in my style, then this should set the width back to that setting. In other words. I want my tabs to be 100px wide so long as there is room. If the control is resized smaller, shrink the tabs down to a MinWidth. Once it's resized larger, each tab should be 100px again

Reply:
I added logic to remember what the previous value was in the tab width.  Take a look.